internal/modules/overview/container/logstreamer.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright (c) 2020 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package container
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	// durContainerUpWait is the time to wait before checking if a container has started
+	durContainerUpWait = 1 * time.Second
+)
+
+type logStreamer struct {
+	namespace  string
+	pod        string
+	containers []string
+	stream     chan LogEntry
+
+	ctx      context.Context
+	cancelFn context.CancelFunc
+	client   kubernetes.Interface
+	wg       sync.WaitGroup
+}
+
+var _ LogStreamer = (*logStreamer)(nil)
+
+// NewLogStreamer returns an instance of a logStream configured to stream logs for the given namespace/pod/container(s).
+func NewLogStreamer(ctx context.Context, client kubernetes.Interface, namespace string, podName string, containerNames ...string) (*logStreamer, error) {
+	ctx, cancelFn := context.WithCancel(ctx)
+
+	if shouldFetchContainerNames(containerNames) {
+		// reset containerNames since it contains only empty entries
+		containerNames = []string{}
+
+		pod, err := client.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
```

Comment:
Is passing `kubernetes.Interface` needed here? A few of these Get requests can be avoided by going through dashConfig.Objectstore

Reply:
We need it for the actual GetLogs and Stream calls that happen in https://github.com/vmware-tanzu/octant/pull/841/files#diff-24e76a2ac8fca0cc788b379f6e54630eR145